pytest_mh/utils/firewall.py

```diff
@@ -0,0 +1,302 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import Literal, TypeAlias
+
+from .. import MultihostHost, MultihostUtility
+from ..ssh import SSHLog
+
+__all__ = ["LinuxFirewalld"]
+
+
+ProtocolSpec: TypeAlias = Literal["tcp", "udp"]
+"""Firewall protocol specification."""
+
+
+PortSpec: TypeAlias = int | tuple[int, ProtocolSpec]
+"""Firewall port specification."""
+
+
+ServiceSpec: TypeAlias = str | PortSpec
+"""Firewall service specification: service name | port | (port, "tcp" | "udp")."""
+
+
+class GenericFirewall(ABC, MultihostUtility):
+    """
+    Configure host firewall.
+
+    All changes are automatically reverted when a test is finished.
+    """
+
+    @abstractmethod
+    def accept(self, what: PortSpec | list[PortSpec]) -> None:
```

Comment:
Can a more descriptive variable name be used other than `what` ? Same applies for all uses of `what` in this file

Reply:
Switch to port (GenericFirewall, WindowsFirewall) and port_and_service (LinuxFirewalld)